File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int n=s.length(),maxfreq=0,ans=0,j=0;
        unordered_map<char,int>m;
        for(int i=0;i<n;i++){
            m[s[i]]++;
            maxfreq=max(maxfreq,m[s[i]]);
            while(i-j+1-maxfreq>k){
                m[s[j]]--;
                j++;
            }
            ans=i-j+1;
        }
    return ans;}
};

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        map={}
        l=0
        maxf=0
        for i in range(len(s)):
            map[s[i]]=1+map.get(s[i],0)
            maxf=max(maxf,map[s[i]])
            
            if (i-l+1)-maxf>k:
                map[s[l]]-=1
                l+=1
        return (i-l+1)
        